registry/remote/example_test.go

```diff
@@ -0,0 +1,312 @@
+			}{
+				Tags: []string{"tag1", "tag2"},
+			}
+			json.NewEncoder(w).Encode(result)
+		case p == fmt.Sprintf("/v2/%s/blobs/uploads/", exampleRepositoryName) && m == "GET":
+			w.WriteHeader(http.StatusCreated)
+		case (p == fmt.Sprintf("/v2/%s/manifests/latest", exampleRepositoryName) || p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleDigest)) && m == "HEAD":
+			w.Header().Set("Content-Type", ocispec.MediaTypeImageLayer)
+			w.Header().Set("Docker-Content-Digest", exampleDigest)
+			w.Header().Set("Content-Length", strconv.Itoa(len([]byte(exampleBlob))))
+		case (p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleDigest)) && m == "GET":
+			w.Header().Set("Content-Type", ocispec.MediaTypeImageLayer)
+			w.Header().Set("Docker-Content-Digest", exampleDigest)
+			w.Header().Set("Content-Length", strconv.Itoa(len([]byte(exampleBlob))))
+			w.Write([]byte(exampleBlob))
+		case p == fmt.Sprintf("/v2/%s/blobs/%s", exampleRepositoryName, exampleDigest) && m == "GET":
+			w.Header().Set("Content-Type", ocispec.MediaTypeImageLayer)
+			w.Header().Set("Docker-Content-Digest", exampleDigest)
+			w.Header().Set("Content-Length", strconv.Itoa(len([]byte(exampleBlob))))
+			w.Write([]byte(exampleBlob))
+		case p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleTag) && m == "PUT":
+			w.WriteHeader(http.StatusCreated)
+		}
+
+	}))
+	defer ts.Close()
+	u, err := url.Parse(ts.URL)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	host = u.Host
+	http.DefaultClient = ts.Client()
+
+	os.Exit(m.Run())
+}
+
+// ExampleRepository_Tags gives example snippets for listing tags in a repository.
+func ExampleRepository_Tags() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	repo.Tags(ctx, func(tags []string) error {
+		for _, tag := range tags {
+			fmt.Println(tag)
+		}
+		return nil
+	})
+
+	// Output:
+	// tag1
+	// tag2
+}
+
+// ExampleRepository_Push gives example snippets for pushing a blob.
+func ExampleRepository_Push() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	mediaType, content := ocispec.MediaTypeImageLayer, []byte("Example blob content") // Setup input: 1) media type and 2)[]byte content
+	desc := ocispec.Descriptor{                                                       // Assemble a descriptor
+		MediaType: mediaType,                 // Set mediatype
+		Digest:    digest.FromBytes(content), // Calculate digest
+		Size:      int64(len(content)),       // Include content size
+	}
+	repo.Push(ctx, desc, bytes.NewReader(content)) // Push the blob
```

Comment:
Can we use the output from push?

Reply:
The push only returns error. I can add handlers for returned error and only output `succeed` when there is no error.